File: ts/project/.sdk/tm/py/core/operation.py
# ProjectName SDK operation

from __future__ import annotations
from utility.voxgig_struct import voxgig_struct as vs


class ProjectNameOperation:
    def __init__(self, opmap=None):
        if opmap is None:
            opmap = {}

        entity = vs.getprop(opmap, "entity")
        if not isinstance(entity, str) or entity == "":
            entity = "_"
        self.entity = entity

        name = vs.getprop(opmap, "name")
        if not isinstance(name, str) or name == "":
            name = "_"
        self.name = name

        inpt = vs.getprop(opmap, "input")
        if not isinstance(inpt, str) or inpt == "":
            inpt = "_"
        self.input = inpt

        self.points = []
        raw_points = vs.getprop(opmap, "points")
        if isinstance(raw_points, list):
            for t in raw_points:
                if isinstance(t, dict):
                    self.points.append(t)

        self.alias = None
        raw_alias = vs.getprop(opmap, "alias")
        if isinstance(raw_alias, dict):
            self.alias = raw_alias

File: ts/project/.sdk/tm/rb/core/operation.rb
# ProjectName SDK operation

require_relative '../utility/struct/voxgig_struct'

class ProjectNameOperation
  attr_accessor :entity, :name, :input, :points, :alias_map

  def initialize(opmap = {})
    opmap ||= {}
    e = VoxgigStruct.getprop(opmap, "entity")
    @entity = (e.is_a?(String) && !e.empty?) ? e : "_"
    n = VoxgigStruct.getprop(opmap, "name")
    @name = (n.is_a?(String) && !n.empty?) ? n : "_"
    i = VoxgigStruct.getprop(opmap, "input")
    @input = (i.is_a?(String) && !i.empty?) ? i : "_"

    @points = []
    raw_points = VoxgigStruct.getprop(opmap, "points")
    if raw_points.is_a?(Array)
      raw_points.each { |t| @points << t if t.is_a?(Hash) }
    end

    raw_alias = VoxgigStruct.getprop(opmap, "alias")
    @alias_map = raw_alias.is_a?(Hash) ? raw_alias : nil
  end
end
